<!DOCTYPE html>
<html lang="en">
<head>
    <title>GeneQuery</title>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <link href="style.css" rel="stylesheet">
</head>
<body>
<div class="content">
    <div class="header-box">
        <h1 class="h1">GeneQuery</h1>
    </div>
    <div class="query-box">
        <div class="row">
            <div class="col fix"><p class="label">Query species</p></div>
            <div class="col fix"><p class="label">Database species</p></div>
            <div class="col"><p class="label">Genes list</p></div>
        </div>
        <div class="row">
            <div class="col fix">
                <div class="radio-options">
                    <label>
                        <input type="radio" name="query" value="hs" checked />
                        <span>Homo Sapiens</span>
                    </label>
                    <label>
                        <input type="radio" name="query" value="mm" />
                        <span>Mus Musculus</span>
                    </label>
                    <label>
                        <input type="radio" name="query" value="rn" />
                        <span>Rattus Norvegicus</span>
                    </label>
                </div>
            </div>
            <div class="col fix">
                <div class="radio-options">
                    <label>
                        <input type="radio" name="db" value="hs" checked />
                        <span>Homo Sapiens</span>
                    </label>
                    <label>
                        <input type="radio" name="db" value="mm" />
                        <span>Mus Musculus</span>
                    </label>
                    <label>
                        <input type="radio" name="db" value="rn" />
                        <span>Rattus Norvegicus</span>
                    </label>
                </div>
            </div>
            <div class="col genes">
                <textarea id="genes" rows="3" name="genes" class="form-control" placeholder="Genes, separated by comma, newline, whitespace or tab. Example: Bhlhe40, Hmox1, Grhpr, Hk1, Jmjd6, Ndst2">
                  Col5a1,Tgm2,Gpc1,Phkg1,Efna1,Ampd3,Tktl1,Pnrc1,Plaur,Glrx,Maff,Serpine1,Cited2,Gapdh,Errfi1,Ets1,Aldoa,Tmem45a,Pdk1,Pgam2,Atf3,Aldoc,Gcnt2,Pgf,Gpc4,Nr3c1,Fbp1,Pgm1,Igfbp3,Tpd52,Ppfia4,Prkcdbp,Ldha,Rbpj,Pkp1,Plac8,Dcn,Igfbp1,Cdkn1c,Slc37a4,Ppargc1a,Ugp2,Csrp2,Bhlhe40,Hmox1,Grhpr,Hk1,Jmjd6,Ndst2,Dpysl4,Hs3st1,Slc2a5,Tiparp,Siah2,P4ha2,Ptrf,Adm,Hspa5,Pfkp,Pfkfb3,Hk2,Sdc3,Gpc3,Eno2,Ddit4,Mxi1,Gys1,Cav1,Srpx,Plin2,Gbe1,Ier3,Slc6a6,Ankzf1,Aldob,Has1,Kdelr3,Anxa2,Bcan,Brs3,Btg1,Irs2,Selenbp1,Cxcr4,Klhl24,Cdkn1a,Sap30,Vhl,Mif,Dtna,Lox,Lxn,Cdkn1b,Map3k1,Ero1l,Myh9,Pdk3,S100a4,Cp,Casp6,Il6,Tnfaip3,Vldlr,Pam,Nedd4l,Adora2b,Cyr61,Noct,Bnip3l,Tpi1,Scarb1,Nagk,Tpbg,Isg20,Hoxb9,Pfkl,Zfp36,Ids,Klf6,Wsb1,Xpnpep1,Bcl2,Ddit3,Ldhc,Pck1,Ndst1,Ppp1r15a,B3galt6,Pgm2,Vegfa,Sult2b1,Stc1,Kif5a,Pgk1,Sdc2,Slc25a1,Rragd,Jun,Fos,Tgfb3,Ilvbl,Ctgf,Gapdhs,Ccng2,Dusp1,Tpst2,Lalba,Ncan,Ndrg1,Inha,F3,Akap12,Tes,Galk1,Pdgfb,Slc2a3,Fosl2,Hexa,Sdc4,Fam162a,Wisp2,Ak4,Chst3,Egfr,Eno1,Rora,Ppp1r3c,Efna3,Car12,Gck,Large,Nfil3,Tgfbi,Kdm3a,Edn2,Bgn,Pygm,Foxo3,Pklr,P4ha1,Gaa,Stbd1,Prkca,Hdlbp,Eno3
                </textarea>
            </div>
            <svg class="next" xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="#777" viewBox="0 0 24 24">
                <path d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"></path>
            </svg>
            <div class="col">
                <button data-query class="primary button">Run</button>
            </div>
        </div>
    </div>

    <div data-results class="results-box"></div>
    <div data-modals></div>

    <div class="info-box">
        <h2 class="h2">About</h2>
        <p>
            GeneQuery is an expression-based phenotype searching engine based on human, mouse,
            and rat microarray experiments. Unlike other tools, like
            <a href="http://david.abcc.ncifcrf.gov/" target="_blank">DAVID</a> or
            <a href="https://www.broadinstitute.org/gsea/msigdb/index.jsp" target="_blank">MsigDB</a>,
            GeneQuery does not use manually curated gene sets associated with a particular pathway
            or perturbation. Instead, we use
            <a href="http://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/" target="_blank">WGCNA</a>
            clustering to obtain a vast database of co-expression
            clusters from available GEO microarray experiments. The data used span X human, Y mouse,
            and Z rat data series that included 8-200 samples, thus covering a substantial fraction of all
            published microarray expression profiling experiments. We believe that this approach is much
            better suited to address biological complexity and variation, and can help generate valuable
            biological hypotheses.
        </p>

        <p>
            To start using GeneQuery, all you need is a list of genes that you have found to be significant in your experiment.
            It can be a list of differentially expressed genes (we suggest using up- and down-regulated signatures separately),
            or candidate regulated genes from a DNase-Seq or a ChIP-Seq experiment.
            GeneQuery accepts the following types of gene IDs: Entrez, universal gene symbol (HGNC/MGNC),
            RefSeq, and Ensembl. It is also natively supports a search using orthologous genes based on
            <a href="https://www.ncbi.nlm.nih.gov/homologene" target="_blank">NCBI Homologene</a> – which means you
            can search a list of human genes against the mouse database or rat,
            and still get meaningful results (in fact, we <b>strongly encourage you</b> to try it).
        </p>

        <p>
            When output is obtained, results are ranked according to adjusted p-value, calculated as Fisher’s p-value with Bonferroni correction.
            Each result corresponds to a matching module, which number is specified. The output also contains name, abstract,
            and links to GEO dataset that was matched. To see if module actually represents certain meaningful group of co-expressed genes,
            click on “heatmap” and find the module number that was matched.
        </p>

        <p>
            The most important part of the output is <b>the module expression heatmap</b>.
            If you see that the expression of the module of interest corresponds the groups according to sample names.
            However, if you see that the expression does not change according to sample names
            (for example, the module is up-regulated in one sample only),
            you might be dealing with an outlier or a case of failed clustering. In the example below,
            this is one of the results returned by hypoxia gene signature. This is an informative hit,
            capturing the transcriptional signature of hypoxic response.
        </p>

        <p>
            In the <a href="/genequery/downloads/">downloads</a> section you can obtain all of
            the databases used by GeneQuery, as well as pre-computed results for all
            of the MsigDB hallmark genesets queried against our human, mouse, and rat databases.
        </p>
    </div>
</div>

<script type="text/template" id="table-row">
    <tr>
        <td>{:name}</td>
        <td>
            <div onclick="ShowHeatmap('{:db}/{:name}', {:module})" class="cell">{:module}</div>
        </td>
        <td>{:log10_apv}</td>
        <td>
            <div onclick="ShowOverlap('{:db}', '{:name}', '{:module}')" class="cell">{:genes}/{:total}</div>
        </td>
        <td>
            <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={:GSE}">{:GSE}</a>
        </td>
        <td>
            <div onclick="DownloadGMT('{:db}', '{:name}')" class="cell cell-center">
                <svg width="1em" height="1em" viewBox="0 0 16 16" class="bi bi-cloud-arrow-down-fill" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
                    <path fill-rule="evenodd" d="M8 2a5.53 5.53 0 0 0-3.594 1.342c-.766.66-1.321 1.52-1.464 2.383C1.266 6.095 0 7.555 0 9.318 0 11.366 1.708 13 3.781 13h8.906C14.502 13 16 11.57 16 9.773c0-1.636-1.242-2.969-2.834-3.194C12.923 3.999 10.69 2 8 2zm2.354 6.854l-2 2a.5.5 0 0 1-.708 0l-2-2a.5.5 0 1 1 .708-.708L7.5 9.293V5.5a.5.5 0 0 1 1 0v3.793l1.146-1.147a.5.5 0 0 1 .708.708z"/>
                </svg>
            </div>
        </td>
    </tr>
</script>

<script type="text/template" id="table">
    <table class="table">
        <thead>
        <tr>
            <th width="90%">Experiment title</th>
            <th>Module</th>
            <th>Log<sub>10</sub>(adj.p<sub>value</sub>)</th>
            <th>Overlap</th>
            <th>NCBI</th>
            <th>GMT</th>
        </tr>
        </thead>
        <tbody>
        {:data use table-row}
        </tbody>
    </table>
</script>

<script type="text/template" id="overlap">
    <div data-genes-lists class="genes-lists">
        <h3>Overlap of input genes with module {:module} of {:name}:</h3>
        <pre><span class="common-genes">{:common}</span> <span class="full-list">{:full}</span></pre>
        <a onclick="FullGenesList()" class="button button-small">Show <span class="f">full list of genes in module</span><span class="o">overlap only</span></a>
    </div>
</script>

<script type="text/template" id="modal">
    <div class="modal-wrapper">
        <div data-modal-window class="modal-window">
            <div onclick="ModalClose()" class="close">✕ Close [esc]</div>
            {:content}
        </div>
        <div onclick="ModalClose()" class="modal-bg"></div>
    </div>
</script>


<script type="text/javascript" src="template.js"></script>
<script type="text/javascript" src="app.js"></script>
</body>
</html>
